### PR TITLE
Rewrite operator init test running with the actual top level cmd

### PR DESCRIPTION
**Pull request by carolynhu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: operator/cmd/mesh/operator_test.go [hunk 1]
```diff
@@ -82,10 +77,14 @@ func TestOperatorInit(t *testing.T) {
 		},
 	}
 
-	l := clog.NewConsoleLogger(os.Stdout, os.Stderr, installerScope)
-	_, gotYAML, err := renderOperatorManifest(rootArgs, &oiArgs.common)
+	cmd := "operator dump --hub " + oiArgs.common.hub
+	cmd += " --tag " + oiArgs.common.tag
+	cmd += " --operatorNamespace " + oiArgs.common.operatorNamespace
```
Comment (morvencao) [inline operator/cmd/mesh/operator_test.go:82]: Nit: suggest to add back the `watchedNamespaces` flag to make sure watching multiple namespaces case is working.

**Comment by carolynhu** [comment]
@morvencao this test is also timeout on my local setup, do you happen to know why?

**Review by morvencao — COMMENTED** [review]
@carolynhu Pls try to run `make gen` to refresh the goldens.

**Comment by carolynhu** [comment]
> @carolynhu Pls try to run `make gen` to refresh the goldens.

it's just to make it pass the presubmit test,  which i will add a commit to update that.

what i mean here is the "timeout" issue...

**Comment by morvencao** [comment]
@carolynhu emm, the `operator init` command need the kube context to run, looks like the it is missing in the `runCommand`: https://github.com/istio/istio/blob/master/operator/cmd/mesh/manifest_shared_test.go#L286-L294

**Pull request closed without merging**